common/utils.go

```diff
@@ -288,3 +290,31 @@ func SaveHostMetadata(metadata Metadata, fileName string) error {
 
 	return err
 }
+
+func GetAzureCloud() (string, error) {
+	req, err := http.NewRequest("GET", azCloudUrl, nil)
```

Comment:
can you add a defer function that logs the error? If for any reason this function returns an error ( for public cloud ), we would disable the telemetry and that would be bad. It would be good to log and know if this functions errors out for any reason, ideally we would add the metric on this error so that we get ICM and look at the error and correct it so that we start getting the logs correctly.

Reply:
im logging the error at calling place..we cannot log metric as this is base condition for sending telemetry

Reply:
one thing we can do is that return error and the calling side can retry.